compiler/rustc_target/src/spec/apple_base.rs

```diff
@@ -173,7 +173,23 @@ pub fn opts(os: &'static str, arch: Arch) -> TargetOptions {
     }
 }
 
-fn deployment_target(var_name: &str) -> Option<(u32, u32)> {
+pub fn deployment_target(target: &Target) -> String {
+    let (major, minor) = match &*target.os {
+        "macos" => {
+            // This does not need to be specific. It just needs to handle x86 vs M1.
+            let arch = if target.arch == "x86" || target.arch == "x86_64" { X86_64 } else { Arm64 };
+            macos_deployment_target(arch)
+        }
+        "ios" => ios_deployment_target(),
+        "watchos" => watchos_deployment_target(),
+        "tvos" => tvos_deployment_target(),
+        _ => unreachable!("unknown Apple target OS"),
+    };
+
+    format!("{major}.{minor}")
+}
+
+fn from_set_deployment_target(var_name: &str) -> Option<(u32, u32)> {
     let deployment_target = env::var(var_name).ok();
     deployment_target
         .as_ref()
```

Comment:
preexisting, but this function deserves a few question mark operators and named bindings to become more readable.

Reply:
It should be better now. Primarily, things are named now to make the parsing more reasonable to read over.